### PR TITLE
refactor: migration to typescript

### PR DESCRIPTION
**Pull request by SeanCassiere**
* Switched to rollup for the building of the package.
* Babel preset for typescript added.
* Added a basic example dir, to be able to view the project in a browser in local dev.

Note: Since Enzyme is dead, in the future should the `devDeps` need to be upgraded to `react^18.0.0`, the testing library would have to be swapped out for something that supports the intricacies of react18.

### CODE REVIEW

**Comment by SeanCassiere** [comment]
Use a squash commit please, since I used some horrible commit-messages 😅

**Comment by SeanCassiere** [comment]
Closes #7

**Comment by lvilasboas** [comment]
@felquis any updates on merging it?

**Comment by felquis** [comment]
Hey @SeanCassiere sorry I couldn't get to test it yet.

I'm going to make an effort to review it first thing in the morning.

cc @lvilasboas

File: jest.config.js [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
+    "<rootDir>/test/__setup__/index.js",
   ],
-  testRegex: '/test/.*?\\.(test|spec)\\.js$',
+  testRegex: "/test/.*?\\.(test|spec)\\.js$",
   collectCoverage: false,
-  collectCoverageFrom: [
-    'src/**/*.{js,jsx}',
-  ],
-  coverageReporters: ['json', 'lcovonly', 'text', 'clover'],
+  collectCoverageFrom: ["src/**/*.{js,jsx}"],
```
Comment (felquis) [inline jest.config.js:19]:
```suggestion
  collectCoverageFrom: ["src/**/*.{js,jsx}", "src/**/*.{ts,tsx}"],
```

**Review by felquis — APPROVED** [review]
I have tests within the examples folder, and on an external project I have setup in localhost, everything is working just fine.

I think it's a good thing to add the .ts and .tsx to code coverage, that's a good feature we should keep around.

Amazing work @SeanCassiere 

:shipit:

**Pull request merged**